### PR TITLE
Added a Case-insensitive search to the table's Header search input that works on json database column.

### PR DESCRIPTION
**Pull request by chamoux**
# ⚡ PowerGrid ⚡ Pull Request

Welcome and thank you for your interest in contributing to our project!. You must use this template to submit a Pull Request or it will not be accepted.

💡 Please complete this template to submit a Pull Request, we cannot accept incomplete Pull Requests.

## Guidelines

`   🗒️ `  Please read the [Contributing Guide](https://github.com/Power-Components/livewire-powergrid/blob/main/CONTRIBUTING.md) and follow all steps listed there.

`   ✍️ `  Give this PR a meaningful title.

`   📣 `  Describe your PR details below.

### Motivation

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

### Related Issue(s)

This PR Fixes the Issue #_____.

### Description

This Pull Request adds a case-insensitive searching based on the column type where it checks if the column type is JSON it performs a case insensitive query or else performs the old query. Previously JSON columns were case-sensitive.

### Contribution Guide

- [x] I have read and followed the steps listed in the [Contributing Guide](https://github.com/Power-Components/livewire-powergrid/blob/main/CONTRIBUTING.md).

### Documentation

 This PR requires [Documentation](https://github.com/Power-Components/powergrid-doc) update?

- [ ] Yes
- [x] No
- [ ] I have already submitted a Documentation pull request.

### CODE REVIEW

**Comment by luanfreitasdev** [comment]
Hello!. Thanks for this contributing.
Can you send any test ?

**Comment by chamoux** [comment]
Test Results:
tests

Testing Table in a laravel project:
https://user-images.githubusercontent.com/29353763/224012840-4af33038-afc9-4161-a9d0-d1363a2799f0.mp4

DB Structure:
DB-Structure

**Comment by luanfreitasdev** [comment]
Hello!. I did a test to ensure that in all databases this change has no influence, but in PostgreeSQL it seems that it is not possible to use LOWER as in mysql. Can you verify this issue so we can move on with the PR?

image

File: src/Helpers/Model.php [hunk 1]
```diff
@@ -101,7 +106,11 @@ public function filterContains(): Model
                         $hasColumn = in_array($field, $columnList, true);
 
                         if ($hasColumn) {
-                            $query->orWhere($table . '.' . $field, SqlSupport::like($query), '%' . $this->powerGridComponent->search . '%');
+                            if (DB::getSchemaBuilder()->getColumnType($table, $field) == 'json' && $query->getModel()->getConnection()->getDriverName() != 'pgsql') {
+                                $query->orWhereRaw('LOWER(`' . $table . '` .`' . $field . '`)' . SqlSupport::like($query) . '?', '%' . strtolower($this->powerGridComponent->search) . '%');
```
Comment (luanfreitasdev) [inline src/Helpers/Model.php:110]:
```php
$query->orWhereRaw('LOWER(`' . $table . '` .`' . $field . '`) ' . SqlSupport::like($query) . '?', '%' . strtolower($this->powerGridComponent->search) . '%');
```

I think if adding space before SqlSupport might work

https://w3resource.com/PostgreSQL/lower-function.php

**Review by luanfreitasdev — CHANGES_REQUESTED** [review]

File: src/Helpers/Model.php (hunk 1 above)
Comment (luanfreitasdev) [reply src/Helpers/Model.php:110]: I think Postgree doesn't recognize the dot(.) this: LOWER(table.field)

**Comment by chamoux** [comment]
It seems in SQL you can use Lower on the entire JSON column but in pgsql you need to use it on a property like so ```lower(column->>property)``` I will commit the spaces and remove the if check since it passed the tests on my end you might want to check after me once more.
Thank you

**Comment by chamoux** [comment]
Second Round of tests:
Tests2
Since ILIKE Already accomplishes what is needed as shown below:
pgsql
we can check the database type if it is different that pgsql then we execute the new query else execute the old one.

**Pull request merged**